### DISCUSSION TITLE
[회의록] : 1주차 장은진

### DISCUSSION THREAD

**Discussion by jihee1103** [FE3-part4-team2/-Nomad#6] [회의록]
### ✏️안건(회의내용)
- [x]  discord web hook연결
- [x] 스토리북은 디자이너와 협업하지 않고 있고 규모가 크지 않기 때문에 꼭 필요하지 않다고 판단하여 사용하기 않기로 결정
- [x] 회의록은 주마다 돌아가면서 github discussion 회의록 탭에 작성하기 => 주마다 한개의 회의록을 생성해서 그 주차의 회의록은 그 안에 작성할 것 / 은진 -> 율민 -> 주영 -> 준형 -> 지희
---
- [x] PR단위는 크지 않게하기, 커밋 5개 이하로 작성하기
- [x] 잊지 않고 아침마다 작업 시작전 PUll받고 시작하기
- [x] main레포지토리로 머지하는거 막기
- [x] default 레포를 dev로 설정하기
- [x] approve인원 설정하기
---
- [x] scss파일 작성시 modul.scss로 작성해야함 ex) nav.module.scss
- [x] variables파일 각 파일마다 import해서 써야지 적용된다
- [x] css 파일명은 camelCase로 작성하기
- [x] css클래스 명은 camelCase로 자유롭게 짓지만, 이해하기 쉽게 작성하기 (module이니까 conflict 걱정 하지 않아도 되기 때문)
- [x] css 반응형 사이즈 기준 정함 => PC: 1200px / Tablet: 768px ~ 1199px / Mobile: 320px ~ 767px
---
- [x] 공통적으로 자주 쓰는 타입은 상위 types 폴더에 작성하기 / 각 컴포넌트에서만 쓰이는 타입은 해당 컴포넌트 파일에 작성하기
- [x] 타입 작성시 interface를 default로 쓸 것
- [x] 타입 이름은 파스칼케이스로 컴포넌트명 + Props로 작성하기 ex) interface NavProps

### ✏️해야할 것
- [x] PR 템플릿 나타나게 하기
- [ ] 로딩 페이지 만들어야함

### ✏️진행사항(프로젝트 작업 진행사항)
- 각자 맡은 컴포넌트 디자인 먼저 완성 후 각자 맡은 페이지 작업 진행하기로 함
- 각자 맡은 컴포넌트 완성한 사람은 PR올리고, 못 한사람은 4/17까지는 완성하기로 함
- 토스트는 _app.tsx 페이지에서 토스트 컨테이너, 토스트 컴포넌트 import해서 토스트 적용하기 편하게 함

**Comment by eundin** [comment]
## 24.04.16.화


### ✏️안건(회의내용)
- [x] 로딩페이지 만들기 =>  기본적인 기능 구현, 디자인 완성을 우선으로 할 필요가 있음. 따라서 어느정도 프로젝트 구현이 된 후에 만들기로함 / 스켈레톤, 스피닝 등 어떤 것을 사용할지 결정필요
- [x] 기존 img태그는 Image 컴포넌트로 사용하자 => 하지만 Next의 Imge 컴포넌트에 대한 이해 필요해서 알아보고 오기로 함
- [x] lazy loading을 쓰면 성능 최적화에 도움이 된다 => 하지만 lazy loading에 대한 이해 필요해서 알아보고 오기로 함
- [x] _app.tsx파일은 기존 React의 App.tsx파일과 같은 역할을 한다고 보면 된다
- [x] navigation bar같이 모든 페이지에 적용돼야 하는 컴포넌트는 layout컴포넌트를 만들어서 적용해야한다
  - Next14의 layout.tsx파일은 Next12에는 적용이 되지 않는다
- [x] react-hook-form 사용하기로 함 / 로그인, 회원가입 페이지, 내정보, 내체험 등록 부분에 적용
- [x] 디스코드랑 figjam에 흩어져 있는 컨벤션 figjam에 모아서 정리하자


### ✏️해야할 것
- [x] PR 템플릿 나타나게 하기
- [x] Next의 Image 컴포넌트에 대해서 알아보고 오기
- [x] Lazy Loading에 대해 알아보고 오기


### ✏️진행사항(프로젝트 작업 진행사항)
- 각자 맡은 컴포넌트 완성한 사람은 PR올렸음
- PR 머지 시 conflict 해결을 위해서
  - approve가 안 된 PR들 중 먼저 올라온 PR부터 approve하고 merge후 -> 그 다음 PR작업 한 사람이 PULL받아서 conflict해결하고 merge하는 방식으로 진행함
- 오늘내로 각자 맡은 공통 컴포넌트 디자인 완성될 예정

**Comment by eundin** [comment]
## 24.04.17.수


### ✏️안건(회의내용)
- [x] rem으로 작성시 크기가 이상하게 적용되는 이슈 있었음 (해결완료)
- [x] 기존 img태그로 작성하던 것들 Image 컴포넌트로 사용하기로 확정 (icon 포함)
- [x] 최적화를 위해 lazy loading 사용하기로함 

### ✏️해야할 것
x

### ✏️진행사항(프로젝트 작업 진행사항)
- 각자 맡은 컴포넌트 작업 완료
- 각자 맡은 페이지 작업 시작

**Comment by eundin** [comment]
## 24.04.18.목


### ✏️안건(회의내용)
- [x] PR올릴때 reviewer 설정하자
- [x] PR올릴 때 close issue해주거나, 우측의 Development부분에서 issue선택해서 닫아줄 수 있다
- [x] 머지하고 나서 브랜치 자동으로 삭제되게 하자
- [x] 멘토님이 PR에 리뷰 남겨주시니 확인한번씩 해보시면 좋을 거 같다

### ✏️해야할 것
- [x] 머지 후 브랜치 자동 삭제하게 설정하기
- [x] PR comment 알림 설정하기

### ✏️진행사항(프로젝트 작업 진행사항)
- 각자 맡은 페이지 작업진행 중

**Comment by eundin** [comment]
## 24.04.19.금


### ✏️안건(회의내용)
- [x] PR머지되면 브랜치 자동으로 삭제하게 했음 => 설정 이전의 삭제 되지 않은 브랜치들은 각자 삭제하기로함
- [x] font만 rem으로 작성하고 나머지는 px로 작성하기
- [x] 1시간이상 막히는 부분이 있으면 멘토님께 도움을 요청해보자
- [x] 회고록 figjam에 적기로 함
- [x] 각자 진행상황 공유했음


### ✏️해야할 것
- [x] 각자 안 쓰는 브랜치 삭제하기
- [ ] 토요일 23:59전까지 figjam에 회고록 적기

### ✏️진행사항(프로젝트 작업 진행사항)
- 각자 맡은 페이지 작업진행 중
